src/sentry/static/sentry/app/views/alerts/details/body.tsx

```diff
@@ -191,15 +191,17 @@ export default class DetailsBody extends React.Component<Props> {
                 <React.Fragment>
                   <SideHeader>
                     <span>{t('Alert Rule')}</span>
-
-                    <SideHeaderLink
-                      to={{
-                        pathname: `/settings/${params.orgId}/projects/${incident?.projects[0]}/alerts/metric-rules/${incident?.alertRule.id}/`,
-                      }}
-                    >
-                      <IconEdit />
-                      {t('View Rule')}
-                    </SideHeaderLink>
+                    {incident?.alertRule &&
+                      incident?.alertRule?.status !== AlertRuleStatus.SNAPSHOT && (
+                        <SideHeaderLink
+                          to={{
+                            pathname: `/settings/${params.orgId}/projects/${incident?.projects[0]}/alerts/metric-rules/${incident?.alertRule.id}/`,
```

Comment:
Will `alertRule` always exist if `incident` is ready?

```suggestion
                            pathname: `/settings/${params.orgId}/projects/${incident?.projects[0]}/alerts/metric-rules/${incident?.alertRule?.id}/`,
```

Reply:
Yeah, I don't think there is any case where an incident's alert rule will not be defined now.